### PR TITLE
Match source and sourcehost case insensitively

### PR DESCRIPTION
**Pull request by niklasnatter**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

This PR adjusts the code to match the `source` and `sourceHost` of a redirect in a case-insensitive manner.

Assuming a redirect with `source: "redirect-url", sourceHost: "www.test.com"`:

Behaviour before: matches only `www.test.com/redirect-url`
Behaviour after: matches `www.test.com/redirect-url`, `www.test.com/redirect-URL`, `www.test.com/REDIRECT-url`, ...

#### Why?

Because [MySQL is case insensitive by default](https://stackoverflow.com/questions/12129571/doctrine2-case-sensitive-query). This means it is not possible to create two redirects that use the same characters (eg. one redirect for `redirect-url` and one redirect for `redirect-URL`). 

Because of this, without this change, it is not possible to redirect both urls (`www.test.com/redirect-url` and `www.test.com/redirect-URL`) at all.

### CODE REVIEW

File: Routing/RedirectRouteProvider.php [hunk 1]
```diff
@@ -49,7 +49,7 @@ public function getRouteCollectionForRequest(Request $request)
         }
 
         $route = new Route(
-            $redirectRoute->getSource(),
+            $pathInfo,
```
Comment (alexander-schranz) [inline Routing/RedirectRouteProvider.php:60]: @nnatter why this change?

Comment (alexander-schranz) [reply Routing/RedirectRouteProvider.php:52]: Think here is missing some of:

```
$redirectRoute = $this->redirectRouteRepository->findEnabledBySource(mb_strtolower($pathInfo), $host)
```

Comment (alexander-schranz) [reply Routing/RedirectRouteProvider.php:60]: Is this the route name? Should we maybe more go with: `'sulu_redirect.redirect_' . $redirectRoute->getId()` or something 🤔

Comment (niklasnatter) [reply Routing/RedirectRouteProvider.php:60]: no it is the `path` of the route. this is used for generating the regex of the route here:
https://github.com/symfony/symfony/blob/252f85c2c249da7d8f2a490b57cc71199d51fcc9/src/Symfony/Component/Routing/RouteCompiler.php#L75-L92

it looks like the regex is evaluated case sensitive. if we pass `/test123` here but the request contains `TEST123`, symfony will not match the request to the route

Comment (alexander-schranz) [reply Routing/RedirectRouteProvider.php:60]: Okay this make sense not sure if we need to use here `$pathInfo` or `$request->getPathInfo` 🤔

**Review by alexander-schranz — APPROVED** [review]
